openstack_plugin/compat.py

```diff
@@ -400,53 +423,91 @@ def _process_security_group_rules(self):
         if rules:
             self.kwargs['security_group_rules'] = rules
 
-    def _map_server_networks_config(self):
+    def _process_operation_inputs(self, openstack_type):
+        """
+        This method will process and the args provided via interface
+        operations and try to merge them if possible to be compatible and
+        consistent with plugin 3.x
+        :param str openstack_type: Openstack object type.
+        """
+        # For all operations, old plugin allow to override the resource_id &
+        # openstack_config via interface operations inputs, so these need to
+        # handle and merge/update with the one provided using node properties
+        openstack_config = self._properties.get('openstack_config', {})
+        if self.kwargs.get('openstack_config'):
+            op_config = self.kwargs.pop('openstack_config')
+            openstack_config.update(op_config)
+
+        if self.kwargs.get('resource_id'):
+            self._properties['resource_id'] = self.kwargs.pop('resource_id')
+
+        # Ignore any deprecated config
+        for item in DEPRECATED_CONFIG:
+            if self.kwargs.get(item):
+                self.kwargs.pop(item)
```

Comment:
Let's say for example someone uses `nova_url`, what should we do? Should we:

- Just remove the key (like right now).
- Add a log like `logger.warn('removed key deprecated client key {0}`? (Probably)
- Try to transform to the correct behavior? (Maybe not).

Reply:
Well, the problem with this, is that there are no mapping for such param like `nova_url` which I think it is a good think we should add a warning message for that , so I would say that adding warning message will be good ?

Reply:
Yeah warn is good or maybe even `error` would be better so a user is certain not to miss it.